components/app.js

```diff
@@ -0,0 +1,348 @@
+    const restaurants = restaurantsData;
+
+    restaurants.forEach(({ name, food }) => {
+        const foodKey = food.sort().join('-'); 
+        if (!foodMap.has(foodKey)) {
+            foodMap.set(foodKey, [name]);
+        } else {
+            foodMap.get(foodKey).push(name);
+        }
+    });
+
+    const similarRestaurants = Array.from(foodMap.values())
+        .filter(restaurants => restaurants.length > 1);
+
+    console.log("Restaurants serving exactly similar foods:");
+    console.log(similarRestaurants);
+}
+
+/**
+ * Identify how many users ordered both food item & beverage
+ */
+function getUsersWithFoodAndBeverage() {
+    const orders = ordersData;
+
+    const usersWithFoodAndBeverage = new Set(
+        orders.filter(order => (
+            order.item && order.item.trim() !== '' && 
+            order.drink && order.drink.trim() !== '' 
+        )).map(order => order.userId)
+    );
+    
+    const numberOfUsers = usersWithFoodAndBeverage.size;
+    console.log(`Number of users who ordered both food and beverage: ${numberOfUsers}`);
+}
+
+/**
+ * Given any restaurant name & item name identify how many sold
+ */
+function getSoldQuantityInfo() {
+    const orders = ordersData;
+    const restaurantName = prompt('Please enter the restaurant name'); //Honest John Pizza
+    const itemName = prompt('Please enter the food item name'); //cheese pizza
+
+    if(restaurantName.length !== 0 || itemName.length !== 0){
+        const totalSold = orders.reduce((total, order) => {
+            if (order.restaurant_name === restaurantName && order.item === itemName) {
+                total += order.quantity;
+            }
+            return total;
+        }, 0);
+    
+        console.log(`Total ${itemName} sold at ${restaurantName}: ${totalSold}`);
+    } else {
+        console.log('Invalid input. Please enter valid restaurant name or item name');
+    }
+    
+}
+
+/**
+ * Which all restaurants serve given beverage or food
+ */
+function getRestaurantInfo() {
+    const restaurantData = restaurantsData;
+    const itemToFind = prompt('Please enter the beverage/food item name'); 
+
+    console.log(itemToFind);
+    if (!itemToFind.length !== 0) {
+        const restaurantsServingItem = restaurantData
+        .filter(restaurant => restaurant.food.includes(itemToFind) || restaurant.beverages.includes(itemToFind))
+        .map(restaurant => restaurant.name);
+
+        console.log(`Restaurants serving ${itemToFind}:`);
+        console.log(restaurantsServingItem);
+    } else {
+        console.log('Invalid input. Please enter valid beverage/food item name');
+    }
+}
+
+/**
+ * Sort orders by date
+ */
+function sortOrdersByDate() {
+    const userInput = prompt('Do you want to sort in ASC/DESC?');
+
+    const orders = ordersData;
+    const ordersWithUpdatedDate = orders.map(obj => ({ ...obj, date: new Date(obj.date) }));
+
+    let sortedOrders;
+    if (userInput === 'ASC' || userInput === 'DESC') {
```

Comment:
can single quotes be from another file

Reply:
I have added the constants to the app-constants file